### PR TITLE
Added JSON support to APK sizing

### PR DESCRIPTION
**Pull request by allisonbm92**
* Protocol buffers are no longer supported. JSON is now used.

* Added tests for the sizing code.

### CODE REVIEW

File: buildSrc/src/main/groovy/com/google/firebase/gradle/ApkSizeJsonBuilder.groovy [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+/** A helper class that generates the APK size measurement JSON report. */
+class ApkSizeJsonBuilder {
+
+    private static final String PULL_REQUEST_TABLE = "PullRequests"
+    private static final String PULL_REQUEST_COLUMN = "pull_request_id"
+    private static final String APK_SIZE_TABLE = "ApkSizes"
+    private static final String SDK_COLUMN = "sdk_id"
+    private static final String APK_SIZE_COLUMN = "apk_size"
+
+    // This comes in as a String and goes out as a String, so we might as well keep it a String
```
Comment (ashwinraghav) [inline buildSrc/src/main/groovy/com/google/firebase/gradle/ApkSizeJsonBuilder.groovy]: 🤣

File: buildSrc/src/main/groovy/com/google/firebase/gradle/ApkSizeJsonBuilder.groovy [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    def toJsonString() {
+        if (sdkSizes.isEmpty()) {
+            throw new IllegalStateException("No sizes were added")
+        }
+
+        def sizeArray = { sdkId, size -> "[$pullRequestNumber, $sdkId, $size]" }
+        def sizeArrayJoiner = new StringJoiner(",")
+
+        sdkSizes.each {
```
Comment (ashwinraghav) [inline buildSrc/src/main/groovy/com/google/firebase/gradle/ApkSizeJsonBuilder.groovy]: I vote to just make this inline

Comment (ashwinraghav) [reply buildSrc/src/main/groovy/com/google/firebase/gradle/ApkSizeJsonBuilder.groovy]: This should work?
```
sdkSizes.collect { size ->
 "[$pullRequestNumber, $sdkId, $size]"
}.join( ",")
```

File: buildSrc/src/main/groovy/com/google/firebase/gradle/ApkSizeTableBuilder.groovy [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+          throw new IllegalStateException("No sizes added")
+        }
+
+        def table = "|------------------        APK Sizes        ------------------|\n"
+        table +=    "|--    project    --|--  build type   --|--  size in bytes  --|\n"
+
+        sdkSizes.each {
+            def line = sprintf("|%-19s|%-19s|%-21s|\n", it.get(0), it.get(1), it.get(2))
+            table += line
+        }
```
Comment (ashwinraghav) [inline buildSrc/src/main/groovy/com/google/firebase/gradle/ApkSizeTableBuilder.groovy]: If you wish for this to be functional style
```
table += sdkSizes.collect { size ->
  sprintf("|%-19s|%-19s|%-21s|", it.get(0), it.get(1), it.get(2))
}.join("\n")
```

File: buildSrc/src/main/groovy/com/google/firebase/gradle/ApkSizeJsonBuilder.groovy (hunk 1 above)
Comment (allisonbm92) [reply buildSrc/src/main/groovy/com/google/firebase/gradle/ApkSizeJsonBuilder.groovy]: I thought keeping a number as a string deserved some explanation :P

File: buildSrc/src/test/groovy/com/google/firebase/gradle/GenerateMeasurementsTaskTest.groovy [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        if (!result.tasks(TaskOutcome.FAILED).isEmpty()) {
+            throw new AssertionError("Smoke test for generate failed", result.getFailure())
+        }
+    }
+
+    @Test
+    public void generate_withTable() {
+        BuildResult result = testProject.build("generate")
+
+        if (!result.tasks(TaskOutcome.FAILED).isEmpty()) {
```
Comment (ashwinraghav) [inline buildSrc/src/test/groovy/com/google/firebase/gradle/GenerateMeasurementsTaskTest.groovy]: I think We should  assert on something more concrete here ?
Maybe the contents of the table as a hardcoded string?

Comment (allisonbm92) [reply buildSrc/src/test/groovy/com/google/firebase/gradle/GenerateMeasurementsTaskTest.groovy]: That's a good idea, but I think we should keep these independent of the output format.

I've added checks that the build did in fact write to the output file or console.

**Pull request merged**